R/tags.R

```diff
@@ -81,12 +81,11 @@ depListToNamedDepList <- function(dependencies) {
 #'
 #' @export
 resolveDependencies <- function(dependencies, resolvePackageDir = TRUE) {
-  # Remove nulls
-  deps <- dependencies[!sapply(dependencies, is.null)]
+  deps <- resolveFunctionalDependencies(dependencies)
 
   # Get names and numeric versions in vector/list form
-  depnames <- sapply(deps, `[[`, "name")
-  depvers <- numeric_version(sapply(deps, `[[`, "version"))
+  depnames <- vapply(deps, function(x) x$name, character(1))
```

Comment:
Doesn't really matter here, but for future reference, it's slightly faster to do it the previous way, because there's one less function call for each element:

```R
vapply(deps, `[[`, "name", FUN.VALUE = character(1))
```

Reply:
Good to know!